src/qililab/platform/platform.py

```diff
@@ -567,6 +600,46 @@ def __str__(self) -> str:
         """
         return str(YAML().dump(self.to_dict(), io.BytesIO()))
 
+    # TODO: determine default average
+    def execute_anneal_program(
+        self, annealing_program_dict: list[dict[str, dict[str, float]]], transpiler: Callable, averages=1
+    ):
+        """Given an annealing program execute it as a qprogram.
+        The annealing program should contain a time ordered list of circuit elements and their corresponging ising coefficients as a dictionary. Example structure:
+
+        .. code-block:: python
+
+            [
+                {"qubit_0": {"sigma_x" : 0, "sigma_y" : 1, "sigma_z" : 2},
+                "coupler_1_0 : {...},
+                },      # time=0ns
+                {...},  # time=1ns
+            .
+            .
+            .
+            ]
+
+        This dictionary containing ising coefficients is transpiled to fluxes using the given transpiler. Then the correspoinding waveforms are obtained and assigned to a bus
+        from the bus to flux mapping given by the runcard.
+
+        Args:
+            annealing_program_dict (list[dict[str, dict[str, float]]]): annealing program to run
+            transpiler (Callable): ising to flux transpiler. The transpiler should take 2 values as arguments (delta, epsilon) and return 2 values (phix, phiz)
+            averages (int, optional): Amount of times to run and average the program over. Defaults to 1.
+        """
+        annealing_program = AnnealingProgram(self, annealing_program_dict)
+        annealing_program.transpile(transpiler)
+        annealing_waveforms = annealing_program.get_waveforms()
```

Comment:
Also, why have we chosen the transpiler to be a Callable and not a class? Now the method `transpile()` belongs to `AnnealingProgram` when it should belong to the `Transpiler` class.

Each class should have a single responsibility, and we should thrive to have a similar structure as circuit and qprogram: the Transpiler class receives an AnnealingProgram and transpiles it.

Reply:
I don't see a problem moving it into a class in the future when we need some more complicated functionality for the transpilaiton process. Right now `AnnealingProgram.transpile()` is just calling the transpiler (`transpiler: Callable` in the arguments) at each time step (so the transpiler _is_ a different class, this is just calling it to transpile the program)

Reply:
`Callable` is not a class though, its a function.

Question: How do we expect users to create/use these transpilers? Do we have a number of predefined algorithms for transpilation?

Reply:
Yes, right now the transpiler is a callable class (we have only a one qubit simple implementation though) https://github.com/qilimanjaro-tech/qililab/blob/main/src/qililab/analog/one_qubit_transpiler.py that takes other functions (methods from fittings, stored in a calibration / transpiler_calibration file, see an example at https://github.com/qilimanjaro-tech/qili-fluqe/blob/f7fe294d2b69b1515f9cbcd87f89523d6ef93bd2/notebooks/single_qubit_2level_algorithm_demo.ipynb)

Reply:
Again, this is kept quite open because we have yet to test any of this in hardware

Reply:
> `Callable` is not a class though, its a function.
> 
> Question: How do we expect users to create/use these transpilers? Do we have a number of predefined algorithms for transpilation?

We should assume for now that on a higher level someone will want to have a predefined algorithm yes, that being the user running the demo. Will have a deeper answer on the full user story by the end of the week, but we should also consider experimental users wanting to use this building up all the necessary pieces without having a pre-defined complete algorithm.

Reply:
Oh now I get it from seeing fluqe code. You do have a class indeed, and you implement the `__call__` method so it is also a `Callable`.

OK, then this is acceptable for now.

Reply:
But we should add a `TODO` to discuss this transpilers structure at some point.